docs/src/docs/asciidoc/user/performance.adoc

```diff
@@ -0,0 +1,138 @@
+This library uses multiple magic tricks to do so. It selects the fastest one for a given environment.
+Make sure of what is used to confirm you can't use a faster way on your platform.
+
+== Serialization
+
+Off-heap, disk and clustering need to serialize keys and values before storing them.
+By default, the Java serialization is used.
+It is well-known for not being the fastest thing around.
+Ehcache uses it because it is supported out of the box.
+However, you can increase performances by provoding your own serializers.
+
+== Copier
+
+By default, on-heap storage stores the entries by reference.
+You might want to use a copier for whatever reason.
+This can be much slower so watch out.
+
+== Loader-Writer
+
+Loader-writer is interesting for many reasons.
+First, it protects you against the Thundering Herd.
+However, it needs to pass through more complicated code to do so.
+
+We are expecting it to be a tiny slower.
+But nothing noticeable enough to prevent you from using it.
+
+== Expiration
+
+A cache with no expiration will always be faster.
+
+=== Time to Live
+
+If you need to set an expiration time, TTL will be the faster one.
+This is because the expiration time is calculated once when the entry is inserted in the cache.
+But it still requires an expiration check at access time.
+
+So you can expect a 2% drop in performance when using TTL. Not bad.
+
+=== Time to Idle
+
+TTI is slower than TTL.
+We need to recalculate the expiration time each time the entry is accessed.
+
+=== Custom
+
+In general, using a custom `ExpiryPolicy` will be the slowest.
+Ehcache uses intrinsics to handle other cases.
+When using a custom policy, you are on your own.
+
+== Allocation rate
+
+Ehcache won't allocate any object during a simple `get()`.
+However, keep in mind that your configuration might do so.
+
+For instance, let's say you define an expiry policy like this.
+
+[source,java,indent=0]
+----
+include::{sourcedir35}/integration-test/src/test/java/org/ehcache/docs/Performance.java[tag=expiryAllocation]
+----
+
+<1> Will instantiate a `Duration` every time an entry is accessed
+
+In this case, putting the `Duration` as a constant would solve the problem.
+
+== Time Source
+
+By default, Ehcache uses a `TimeSource` that will retrieve the system time at every call. It is fast but not super
+duper fast. But it is super duper accurate.
+
+You can trade accuracy for speed by using a `TickingTimeSource`. Please read the javadoc for details but the concept is
+that a timer will increase time instead of always retrieving system time.
+
+Switching to `TickingTimeSource`, even with a granularity of 1ms can improve the performance of a `get` of a high as 30%.
+
+The drawback is that a timer will continuously run.
+Also, time might drift from the real time a bit.
+Especially if the granularity or the systemUpdatePeriod is big.
+But in most cases, the drifting doesn't matter much.
```

Comment:
I would like to see little explanation why _doesn't matter much._...
Is it because Ehcache mostly uses timestamps to calculate durations so the "base" time and drifting has little impact ?